### ISSUE TITLE
Grouped convolutions giving different output than separate convs on Windows 10 with cuda

### ISSUE THREAD

**Issue by kampelmuehler** [pytorch/pytorch#37505]
## 🐛 Bug

Grouped Conv2d should give the exact same result as separate Conv2ds, but don't. Works on Linux, but doesn't on Windows 10.

## To Reproduce

```
import torch
import torch.nn as nn

torch.manual_seed(1)

# some random input
a = torch.Tensor(1, 2, 1, 1).random_().cuda()

# 2 individual Conv2d layers
conv1 = nn.Conv2d(1, 1, 3, padding=1, bias=False).cuda()
conv2 = nn.Conv2d(1, 1, 3, padding=1, bias=False).cuda()

# Conv2d layer with groups
conv_gr = nn.Conv2d(2, 2, 3, groups=2, padding=1, bias=False).cuda()
# taking weights from the individual layers
conv_gr.weight.data = torch.cat([conv1.weight.data, conv2.weight.data], 0)

# concatenation of the outputs of the 2 individual layers
print(torch.cat([conv1(a[:, 0:1, ...]), conv2(a[:, 1:2, ...])], 0).data)
# out of layer with groups
print(conv_gr(a).data)
```

With cuda the outputs aren't equal:

```
tensor([[[[-871647.5000]]],
        [[[-140616.8438]]]], device='cuda:0')
tensor([[[[-871647.5000]],
         [[-140616.8594]]]], device='cuda:0')
```

## Expected behavior

Should give the same output as with cpu (or on Linux with cuda for that matter):
```
tensor([[[[-871647.5000]]],
        [[[-140616.8594]]]])
tensor([[[[-871647.5000]],
         [[-140616.8594]]]])
```

## Environment
PyTorch version: 1.5.0
Is debug build: No
CUDA used to build PyTorch: 10.2

OS: Microsoft Windows 10 Pro
GCC version: Could not collect
CMake version: Could not collect

Python version: 3.8
Is CUDA available: Yes
CUDA runtime version: 10.2.89
GPU models and configuration: GPU 0: GeForce GTX 1070
Nvidia driver version: 445.87
cuDNN version: Could not collect

Versions of relevant libraries:
[pip] numpy==1.18.1
[pip] torch==1.5.0
[conda] blas                      1.0                         mkl  
[conda] cudatoolkit               10.2.89              h74a9793_0  
[conda] mkl                       2020.0                      166  
[conda] mkl-service               2.3.0            py38hb782905_0  
[conda] mkl_fft                   1.0.15           py38h14836fe_0  
[conda] mkl_random                1.1.0            py38hf9181ef_0  
[conda] numpy                     1.18.1           py38h93ca92e_0  
[conda] numpy-base                1.18.1           py38hc3f5095_1  
[conda] pytorch                   1.5.0           py3.8_cuda102_cudnn7_0    pytorch

cc @ngimel @peterjc123 @nbcsm @guyang3532

**Comment by pbelevich** [comment]
Could not reproduce

```
In [11]: print(torch.cat([conv1(a[:, 0:1, ...]), conv2(a[:, 1:2, ...])], 0).data)
tensor([[[[-871647.5000]]],


        [[[-140616.8594]]]], device='cuda:0')

In [12]: print(conv_gr(a).data)
tensor([[[[-871647.5000]],

         [[-140616.8594]]]], device='cuda:0')
```

## Environment
PyTorch version: 1.5.0
Is debug build: No
CUDA used to build PyTorch: 10.2

OS: Microsoft Windows 10 Pro
GCC version: Could not collect
CMake version: Could not collect

Python version: 3.8
Is CUDA available: Yes
CUDA runtime version: 10.2.89
GPU models and configuration: GPU 0: GeForce RTX 2080 with Max-Q Design
Nvidia driver version: 445.87
cuDNN version: Could not collect

Versions of relevant libraries:
[pip] numpy==1.18.1
[pip] torch==1.5.0
[pip] torchvision==0.6.0
[conda] blas                      1.0                         mkl
[conda] mkl                       2020.0                      166
[conda] mkl-service               2.3.0            py38hb782905_0
[conda] mkl_fft                   1.0.15           py38h14836fe_0
[conda] mkl_random                1.1.0            py38hf9181ef_0
[conda] pytorch                   1.5.0           py3.8_cuda102_cudnn7_0    pytorch
[conda] torchvision               0.6.0                py38_cu102    pytorch

**Comment by ngimel** [comment]
Grouped and non-grouped convolutions use different implementations, there's no guarantee that results would be bitwise-equivalent. The differences you are getting are on the order of fp accuracy, so it's not a bug.
